cli/compose/convert/service.go

```diff
@@ -308,11 +310,24 @@ func convertServiceConfigObjs(
 			return nil, err
 		}
 
-		file := swarm.ConfigReferenceFileTarget(obj.File)
-		refs = append(refs, &swarm.ConfigReference{
-			File:       &file,
-			ConfigName: obj.Name,
-		})
+		// if the obj.File is identical to the zero-value of
```

Comment:
As above, is this a bug waiting to happen?

Reply:
Possibly, but the thing is I'm not sure under what case we'll add another non-file ConfigReference type, so I can't really code defensively against it, and even if I do, it's unlikely that something else will even be implemented at any point.

Reply:
Off the top of my head, a value explicitly for runtime targets.

Reply:
Yes, it would make sense lol

I've made this change, and added tests to boot.